packages/wpcom.js/src/index.js

```diff
@@ -53,23 +52,12 @@ export default function WPCOM( token, reqHandler ) {
 		this.token = token;
 	}
 
-	// Set default request handler
-	if ( ! reqHandler ) {
-		debug( 'No request handler. Adding default XHR request handler' );
-
-		this.request = function( params, fn ) {
-			params = params || {};
+	const noHandler = () => {
+		debug( 'No request handler. Failing.' );
+		throw new Error( 'No request handler provided' );
```

Comment:
Maybe calling the `fn` callback with error is even better?

Reply:
I'm afraid I don't follow. Which `fn`?

Reply:
Oh, wait, I think I got it :) Done.

Reply:
Yes! The idea is that the handler communicates with the caller not through a return value, but by calling a callback.

Return value is ignored and a thrown exception could be ignored and swallowed, too:
```js
setTimeout( () => {
  request( ( err, data ) => {
    if ( err ) {
      console.error( err );
    }
  } );
  throw 'error'; // this will never be caught, and neither will exception from `request`
}, 1000 );
```

But calling `fn( err )` sends the error to the flow channel that is supposed to handle it.